traits/trait_list_object.py

```diff
@@ -60,33 +60,97 @@ def __repr__(self):
 
 def _normalize_index(index, length):
     """ Normalize integer index to range 0 to len (inclusive). """
+
```

Comment:
Unrelated code change. But since this drew my eye, I also realise that we don't really need `_normalize_index` any more. It's only used for `pop` and `insert`, and in the `pop` case the logic here doesn't match the index manipulation that Python does anyway, so there's no shared logic between those two methods.

So we should probably just inline this in the `insert` method, and add the right simpler logic in `pop`.

Reply:
Done.